xla/backends/cpu/runtime/onednn_thunk.h

```diff
@@ -0,0 +1,55 @@
+/* Copyright 2024 The OpenXLA Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef XLA_SERVICE_CPU_RUNTIME_ONEDNN_THUNK_H_
+#define XLA_SERVICE_CPU_RUNTIME_ONEDNN_THUNK_H_
+
+#if defined(INTEL_MKL) && defined(ENABLE_ONEDNN_V3)
```

Comment:
can we instead use build rules to conditionally exclude sources and headers to avoid ifdefs, like here: https://github.com/openxla/xla/blob/5ad6651eeefd4d0e70990f8ae8efb509f310ddbb/xla/backends/gpu/collectives/BUILD#L116-L117